Setting-Up-Ubuntu-For-Programming.md

```diff
@@ -0,0 +1,219 @@
+# Setting up Ubuntu for Programming
+
+> Note: We will be referring to Ubuntu 16.04 LTS (Xenial Xerus) in this article.
+
+1. Download and Install Ubuntu Desktop
+2. Customising Ubuntu
+3. Jazzing up the Terminal
+4. Installing DevTools and modern web browsers
+5. Downloading Text Editors and IDEs
+6. Alternatives to popular Windows and Mac software
+
+## Download and Install Ubuntu Desktop
+
+You can download the latest version of Ubuntu Desktop [here](http://www.ubuntu.com/download/desktop).
+If you need any help with installing the operating system [this](http://www.ubuntu.com/download/desktop/install-ubuntu-desktop) tutorial by Ubuntu will help you out. If you want to dual boot your computer with Windows and Ubuntu checkout [this](https://help.ubuntu.com/community/WindowsDualBoot) helpful article.
+
+Once Ubuntu is installed, open the terminal and type `sudo apt-get update && sudo apt-get upgrade` to update your packages.
+
+## Customising Ubuntu
+
+#### `.deb` Package Installer
+
+Install gdebi (`.deb` package installer) by typing `sudo apt-get install gdebi`.
+
+#### Removing Bloatware
+
+To remove all pre-installed bloatware because of privacy concerns or to keep your operating system minimal, checkout [this](https://gist.github.com/ansell/61313400e26cd42289f8) gist.
+
+#### Aliases
+
+You can create a temporary alias like this:
+```bash
+alias alias_name="command_to_run"
+```
+However, when you close your shell session, this alias will cease to exist.
+
+To create a permanent alias you will need to edit the `~/.bash_aliases` file. Once you've opened this file with your text editor of choice, add a line at the bottom of the document, similar to the example above.
+
+To find out more, DigitalOcean has a great tutorial that can be found [here](https://www.digitalocean.com/community/tutorials/an-introduction-to-useful-bash-aliases-and-functions).
+
+#### Unity Tweak Tool
```

Comment:
Maybe tweaking and jazzing up the terminal could be separate tutorials or included for a more intermediate tutorial. I think keeping things as close to standard is probably for the best.  When the new user runs into trouble it is best if what is on his screen looks like the picture on the troubleshooting guide hes going to read.